cmd/minishift/cmd/start.go

```diff
@@ -409,26 +409,75 @@ func exportContainerImages(hostVm *host.Host, version string) {
 	fmt.Println(fmt.Sprintf("-- Exporting of OpenShift images is occuring in background process with pid %d.", exportCmd.Process.Pid))
 }
 
-// calculateDiskSizeInMB converts a human specified disk size like "1000MB" or "1GB" and converts it into Megabits
-func calculateDiskSizeInMB(humanReadableDiskSize string) int {
-	diskSize, err := units.FromHumanSize(humanReadableDiskSize)
+type sizeValidatorFunc func(name string, diskSize string) error
+type sizeConverterFunc func(input int64) int
+
+func calculateHumanSize(sizeValidator sizeValidatorFunc, name string, humanReadableSize string, unit string, sizeConverter sizeConverterFunc) (int, error) {
+	if stringUtils.HasOnlyNumbers(humanReadableSize) {
+		humanReadableSize += unit
+	}
+
+	err := sizeValidator(name, humanReadableSize)
+	if err != nil {
+		return -1, err
+	}
+
+	size, err := units.FromHumanSize(humanReadableSize)
 	if err != nil {
+		return -1, err
+	}
+
+	return sizeConverter(size), nil
+}
+
+func calculateMemorySize(humanReadableSize string) int {
+	converter := func(input int64) int {
```

Comment:
Would it not be easier and actually cleaner to let _calculateHumanSize_ just return the amount of memory (in whatever unit you want) and then apply the conversion directly in this function. 

Passing this converter function let's me as reader with several questions in my mind what's going to happen with this function. Where and when and how it is used for example. The amount of re-use you get in _calculateHumanSize_ does imo not outweigh the additional complexity.